### ISSUE TITLE
Model/RecurrencePattern:getDaysOfWeek() results in a ErrorException

### ISSUE THREAD

**Issue by Zombaya** [microsoftgraph/msgraph-sdk-php#76]
When trying to execute `getDaysOfWeek()` on a `Model/RecurrencePattern` you will receive a `ErrorException: Array to string conversion`.

This is produced because the code is trying to cast it to a single `DayOfWeek` instead of an array of `DayOfWeek`-objects.

## Unit test

I created a unit test for this in [a new branch](/integr8rs/msgraph-sdk-php/tree/patch-4).

## Example value of DaysOfWeek
```php
$this->_propDict['daysOfWeek'] = array('monday','wednesday');
```

## Location of the bug
```php
// src/Model/RecurrencePattern.php:L146-L163

/**
* Gets the daysOfWeek
* A collection of the days of the week on which the event occurs. Possible values are: sunday, monday, tuesday, wednesday, thursday, friday, saturday.
* If type is relativeMonthly or relativeYearly, and daysOfWeek specifies more than one day, the event falls on the first day that satisfies the pattern.
* Required if type is weekly, relativeMonthly, or relativeYearly.
*
* @return DayOfWeek The daysOfWeek
*/
public function getDaysOfWeek()
{
    if (array_key_exists("daysOfWeek", $this->_propDict)) {
        if (is_a($this->_propDict["daysOfWeek"], "Microsoft\Graph\Model\DayOfWeek")) {
            return $this->_propDict["daysOfWeek"];
        } else {
            $this->_propDict["daysOfWeek"] = new DayOfWeek($this->_propDict["daysOfWeek"]);
            return $this->_propDict["daysOfWeek"];
        }
    }
    return null;
}
``` 
[AB#7577](https://microsoftgraph.visualstudio.com/0985d294-5762-4bc2-a565-161ef349ca3e/_workitems/edit/7577)

**Comment by RonyDimebag** [comment]
Hi, is there a fix for this?
I just stumbled across this bug, was weird for me on the first place it returns an enum to represent an array as returned by the API.

**Comment by RonyDimebag** [comment]
A small fix locally solved the problem - 

```
public function getDaysOfWeek()
{
    if (array_key_exists("daysOfWeek", $this->_propDict)) {
        return $this->_propDict["daysOfWeek"];
    }
    return null;
}
```

**Comment by Zombaya** [comment]
I don't believe there is a fix for this yet.  And since it's a part of a Model, it will need to be changed in the [codegenerator](https://github.com/microsoftgraph/msgraph-sdk-code-generator).

There is no issue for it yet on that project, and I myself do not know how that project works.  

My solution for now was to just request the data once in a function that transforms this model to another model we use internally.

The fix supplied by @RonyDimebag will not convert the values to a `DayOfWeek`-object, which IMHO would be preferable.

**Comment by RonyDimebag** [comment]
Indeed this is a quick and simple fix to simply allow you get the data as an array, and avoiding exceptions.
Please share your fix if it uses `DayOfWeek` model.

**Comment by rodrigoguariento** [comment]
I had a similar problem with type of object (content) in the same point, into getDaysOfWeek(). I've solved with the following:

	/**
	* RecurrencePattern class
	*
	* @category  Model
	* @package   Microsoft.Graph
	* @copyright © Microsoft Corporation. All rights reserved.
	* @license   https://opensource.org/licenses/MIT MIT License
	* @version   Release: 1.4.0
	* @link      https://graph.microsoft.io/
	*/

    // line: 152
    public function getDaysOfWeek()
    {
        if (array_key_exists("daysOfWeek", $this->_propDict)) {
            if (is_a($this->_propDict["daysOfWeek"], "Microsoft\Graph\Model\DayOfWeek")) {
                return $this->_propDict["daysOfWeek"];
            } else {
                if (gettype($this->_propDict["daysOfWeek"]) == 'string') {
                    return new DayOfWeek($this->_propDict["daysOfWeek"]);
                }
                return $this->_propDict["daysOfWeek"];
            }
        }
        return null;
    }

**Comment by isvargasmsft** [comment]
Hello @Zombaya, 

Thank you very much for your patience. This issue has been resolved in our [new SDK version 2.0.0-RC5](https://github.com/microsoftgraph/msgraph-sdk-php/issues/943).

Please feel free to give it a try and share feedback with us. 

Kind regards, 

Isaac Vargas
